aptos-move/aptos-vm-types/src/change_set.rs

```diff
@@ -710,6 +742,32 @@ impl VMChangeSet {
         Ok(())
     }
 
+    fn squash_additional_group_reads_needing_exchange<K>(
+        group_reads_needing_exchange: &mut BTreeMap<StateKey, (WriteOp, u64)>,
+        additional_group_reads_needing_exchange: BTreeMap<StateKey, (WriteOp, u64)>,
+        skip: &BTreeMap<StateKey, K>,
+    ) -> anyhow::Result<(), VMStatus> {
+        for key in skip.keys() {
+            group_reads_needing_exchange.remove(key);
+        }
+        for (key, additional_value) in additional_group_reads_needing_exchange.into_iter() {
+            if skip.contains_key(&key) {
+                continue;
+            }
+            match group_reads_needing_exchange.entry(key) {
```

Comment:
isn't this whole block equivalent to 

let res = group_reads_needing_exchange.insert(key, additional_value);
// assert on res if we want to keep it.

Reply:
Changed it. We are not doing the assert anymore. I hope it's fine?